package/MDAnalysis/core/topology.py

```diff
@@ -348,3 +350,64 @@ def add_TopologyAttr(self, topologyattr):
         self.attrs.append(topologyattr)
         topologyattr.top = self
         self.__setattr__(topologyattr.attrname, topologyattr)
+
+    def to_json(self, filename):
+        """Write out Topology to JSON form.
+
+        Parameters
+        ----------
+        filename : str
+            File to write.
+
+        """
+        jstruct = {'n_atoms': self.n_atoms,
+                   'n_residues': self.n_residues,
+                   'n_segments': self.n_segments,
+                   'atoms->residues': self.tt.AR.tolist(),
+                   'residues->segments': self.tt.RS.tolist(),
+                   'topattrs': dict()}
+
+        # TODO: add warning that could not serialize some topology attributes
+        for topattr in self.attrs:
+
+            # these don't hold any data of their own
+            if isinstance(topattr, (Atomindices, Resindices, Segindices)):
```

Comment:
What about having `{Atom,Res,Seg}indices._to_json` raise an `AttributeError`? Either by not having them defined or by having them explicitely raise the exception? Then, these attributes will be ignored thanks to the `try` bellow. This would save us this test, avoid special cases, and avoid a `continue` that can be error prone if this loop get extended in the future.